From 8022911b48032f3457f3e062ec11a28b8f6070a0 Mon Sep 17 00:00:00 2001
From: Efe Mert Demir <efemert.demir@idealworks.com>
Date: Mon, 16 Dec 2024 15:35:18 +0000
Subject: [PATCH] Bump the static allocator size for memory_tools_interpose

The original value is way outdated and no longer allows for tests to
run. This should be handled properly by crashing on static allocation
failure instead of proceeding.
---
 osrf_testing_tools_cpp/src/memory_tools/impl/linux.cpp | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/osrf_testing_tools_cpp/src/memory_tools/impl/linux.cpp b/osrf_testing_tools_cpp/src/memory_tools/impl/linux.cpp
index cf69e73..63ff3ee 100644
--- a/osrf_testing_tools_cpp/src/memory_tools/impl/linux.cpp
+++ b/osrf_testing_tools_cpp/src/memory_tools/impl/linux.cpp
@@ -42,9 +42,11 @@ find_original_function(const char * name)
   return original_function;
 }
 
-// An amount of memory that is greater than what is needed for static initialization
-// for any test we run. It was found experimentally on Ubuntu Linux 16.04 x86_64.
-static constexpr size_t STATIC_ALLOCATOR_SIZE = 0x800000;
+// An amount of memory that is greater than what is needed for static
+// initialization for any test we run.
+// Updated in December 2024. The original value of 0x800000 was no longer enough
+// for ROS2 tests.
+static constexpr size_t STATIC_ALLOCATOR_SIZE = 0x2000000;
 using osrf_testing_tools_cpp::memory_tools::impl::StaticAllocator;
 using StaticAllocatorT = StaticAllocator<STATIC_ALLOCATOR_SIZE>;
 static uint8_t g_static_allocator_storage[sizeof(StaticAllocatorT)];
-- 
2.45.2

